x-pack/qa/third-party/active-directory/src/test/java/org/elasticsearch/xpack/security/authc/ldap/ActiveDirectoryGroupsResolverTests.java

```diff
@@ -53,6 +64,33 @@ public void testResolveSubTree() throws Exception {
                 containsString("Supers")));
     }
 
+    @SuppressWarnings("unchecked")
+    public void testResolveTokenGroupsSID() throws Exception {
+        Settings settings = Settings.builder()
+            .put("xpack.security.authc.realms.active_directory.ad.group_search.base_dn", "DC=ad,DC=test,DC=elasticsearch,DC=com")
+            .put("xpack.security.authc.realms.active_directory.ad.domain_name", "ad.test.elasticsearch.com")
+            .put("path.home", createTempDir())
+            .put(RealmSettings.getFullSettingKey(REALM_ID, RealmSettings.ORDER_SETTING), 0)
+            .put(getFullSettingKey(REALM_ID, LdapMetadataResolverSettings.ADDITIONAL_METADATA_SETTING), "tokenGroups")
+            .build();
+        RealmConfig config = new RealmConfig(REALM_ID, settings, TestEnvironment.newEnvironment(settings), new ThreadContext(settings));
+        LdapMetadataResolver resolver = new LdapMetadataResolver(config, true);
+        Map<String, Object> groupSIDs = resolve(null, resolver);
```

Comment:
`groupSIDs` -- > metadata or something similar that indicates that this the user metadata ( that contain tokenGroups )

Reply:
Done